music21/metadata/__init__.py

```diff
@@ -142,115 +238,376 @@ class Metadata(base.Music21Object):
 
-        # a dictionary of Text elements, where keys are work id strings
-        # all are loaded with None by default
-        self._workIds = OrderedDict()
-        for abbreviation, workId in self.workIdAbbreviationDict.items():
-            # abbreviation = workIdToAbbreviation(id)
-            if workId in keywords:
-                self._workIds[workId] = Text(keywords[workId])
-            elif abbreviation in keywords:
-                self._workIds[workId] = Text(keywords[abbreviation])
-            else:
-                self._workIds[workId] = None
+    def setCustom(self, key: str, value: t.Union[t.Any, t.List[t.Any]]):
+        self._set(key, value, isCustom=True)
+
+    def getAllNamedValues(self, skipContributors=False) -> t.List[t.Tuple[str, t.Any]]:
+        '''
+        Returns all values stored in this metadata as a list of (nsKey, value) tuples.
+        nsKeys with multiple values will appear multiple times in the list (rather
+        than appearing once, with a value that is a list of values).
+        The tuple's first element is either of the form 'namespace:name', or a
+        custom key (with no form at all).
+
+        >>> md = metadata.Metadata()
+        >>> md.add('composer', 'Jeff Bowen')
+        >>> md.add('librettist', 'Hunter Bell')
+        >>> md.add('title', '[title of show]')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> all = md.getAllNamedValues()
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Jeff Bowen>),
+        ('marcrel:LBT', <music21.metadata.primitives.Contributor librettist:Hunter Bell>),
+        ('dcterms:title', <music21.metadata.primitives.Text [title of show]>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        >>> all = md.getAllNamedValues(skipContributors=True)
+        >>> all
+        [('dcterms:title', <music21.metadata.primitives.Text [title of show]>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        '''
+        allOut: t.List[t.Tuple[str, t.Any]] = []
+
+        for nsKey, value in self._metadata.items():
+            if skipContributors and self._isContributorNSKey(nsKey):
+                continue
+
+            for v in value:
+                allOut.append((nsKey, v))
+
+        return allOut
+
+    def getAllContributorNamedValues(self) -> t.List[t.Tuple[str, t.Any]]:
+        '''
+        Returns all contributors stored in this metadata as a list of (nsKey, value) tuples.
+        The tuple's first element will be of the form 'namespace:name'.
+
+        >>> md = metadata.Metadata()
+        >>> md.addCustom('composer', 'Not a contributor')
+        >>> md.add('title', '[title of show]')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> md.add('composer', ['Ludwig von Beethoven', 'Wolfgang Amadeus Mozart'])
+        >>> md.add('librettist', 'Marie Červinková-Riegrová')
```

Comment:
again, `[title of show]` is not a good example.  And neither Beethoven nor Mozart ever had an opera with libretto by Marie Červinková-Riegrová .  Karel Šebor would be a better example here.

I know that this can be frustrating, but all music21 docs need to relate to both programming and music researcher audiences and have to pass both group's standards for authenticity.

Reply:
Thanks for explaining, I will do my best.

Reply:
All fixed (I think).

Reply:
Great.